Shoko.Server/API/Startup.cs

```diff
@@ -54,6 +57,28 @@ public Startup(IWebHostEnvironment env)
 
     public void ConfigureServices(IServiceCollection services)
     {
+        services.AddSingleton<IConnectivityMonitor>(_ => new GeneralConnectivityMonitor());
+        
+        services.AddQuartz(q =>
+        {
+            // as of 3.3.2 this also injects scoped services (like EF DbContext) without problems
+            q.UseMicrosoftDependencyInjectionJobFactory();
+
+            // Register the connectivity monitor job with a trigger that executes every 5 minutes
+            q.ScheduleJob<ConnectivityMonitorJob>(
+                trigger => trigger.WithCronSchedule("0 */5 * * * ?").StartNow(), 
+                j => j.WithIdentity(ConnectivityMonitorJob.Key).DisallowConcurrentExecution().Build());
+
+            q.AddJob<ImportJob>(j => j.WithIdentity(ImportJob.Key).DisallowConcurrentExecution().StoreDurably().Build()); // TODO: Maybe add schedule
+            q.AddJob<ScanFolderJob>(j => j.WithIdentity(ScanFolderJob.Key).DisallowConcurrentExecution().StoreDurably().Build()); // TODO: Maybe add schedule
+        });
+
+        services.AddQuartzServer(options =>
+        {
+            // when shutting down we want jobs to complete gracefully
+            options.WaitForJobsToComplete = true;
```

Comment:
TODO: Add cancellation token support through the pipeline, so can can cancel and shutdown in a safe and timely manner.

Reply:
I thought that that was already achieved by this option without the need to pass a cancellation token?

Reply:
We have a static bool, but it's not used everywhere, and we should probably be using a cancellation token anyway

Reply:
The TODOs are not necessarily for you, just things that should happen to improve shoko